### PR TITLE
support setting edition with an environment variable

### LINKED ISSUE #1384: Support setting the value of testthat edition via an environment variable
Hi team!

I hope this finds you well, and thanks for all of the great work on `testthat`.

We use a build system to support R at Google, which unfortunately makes it somewhat difficult to access a package's description file when running tests. Instead, we've modified `testthat` internally to be able to set the edition using an environment variable.

Is this something worth supporting for other users?

Thanks!

### PR DESCRIPTION
**Pull request by michaelquinn32**
Fixes #1384

### CODE REVIEW

File: R/edition.R [hunk 1]
```diff
@@ -58,14 +62,16 @@ edition_name <- function(x) {
...
 local_edition <- function(x, .env = parent.frame()) {
-  stopifnot(is.numeric(x) && length(x) == 1)
+  stopifnot(is_zap(x) || (is.numeric(x) && length(x) == 1))
   old <- edition_set(x)
   withr::defer(edition_set(old), envir = .env)
 }
 
 edition_set <- function(x) {
   if (is_zap(x)) {
+    old <- env_get(testthat_env, "edition", default = 2L)
```
Comment (hadley) [inline R/edition.R:72]: Are sure it makes sense to set a default here?

Comment (michaelquinn32) [reply R/edition.R:72]: TBH, didn't think that much about it.

In my tests, if no default is needed, we can simplify the whole thing to

```
edition_set <- function(x) {
  env_poke(testthat_env, "edition", x)
}
```

This still works with `zap()` and keeps a consistent return type. How does that sound?

Comment (hadley) [reply R/edition.R:72]: Yeah, that makes sense. I probably wrote this for an earlier version of `env_poke()`

Comment (michaelquinn32) [reply R/edition.R]: Thanks!

**Pull request merged**